src/AccessibilityInsights.SharedUx/Controls/CustomControls/AutomatedChecksCustomListControl.xaml.cs

```diff
@@ -0,0 +1,689 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using AccessibilityInsights.CommonUxComponents.Controls;
+using AccessibilityInsights.CommonUxComponents.Dialogs;
+using AccessibilityInsights.Extensions.Helpers;
+using AccessibilityInsights.Extensions.Interfaces.IssueReporting;
+using AccessibilityInsights.SharedUx.Enums;
+using AccessibilityInsights.SharedUx.FileIssue;
+using AccessibilityInsights.SharedUx.Highlighting;
+using AccessibilityInsights.SharedUx.Telemetry;
+using AccessibilityInsights.SharedUx.Utilities;
+using AccessibilityInsights.SharedUx.ViewModels;
+using Axe.Windows.Core.Bases;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Windows;
+using System.Windows.Automation;
+using System.Windows.Automation.Peers;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Data;
+using System.Windows.Input;
+using System.Windows.Media;
+
+namespace AccessibilityInsights.SharedUx.Controls.CustomControls
+{
+    /// <summary>
+    /// Interaction logic for AutomatedChecksCustomListControl.xaml
+    /// </summary>
+    public partial class AutomatedChecksCustomListControl : UserControl
+    {
+        /// <summary>
+        /// This width will limit the element path column to approx. 75 chars
+        /// </summary>
+        const int MaxElemPathColWidth = 360;
+
+        private bool _allExpanded;
+
+        private CheckBox CheckBoxSelectAll => this.chbxSelectAll;
+
+        /// <summary>
+        /// Whether there is a screenshot available (if not, checkboxes should be disabled)
+        /// </summary>
+        private bool ScreenshotAvailable => _controlContext?.DataContext?.Screenshot != null;
+
+        private AutomatedChecksCustomListControlContext _controlContext;
+
+        /// <summary>
+        /// Currently selected items
+        /// </summary>
+        internal IList<RuleResultViewModel> SelectedItems { get; } = new List<RuleResultViewModel>();
+
+        public AutomatedChecksCustomListControl()
+        {
+            InitializeComponent();
+            AddHandler(Thumb.DragDeltaEvent, new DragDeltaEventHandler(Thumb_DragDelta), true);
+        }
+
+        internal void SetControlContext(AutomatedChecksCustomListControlContext controlContext)
+        {
+            _controlContext = controlContext ?? throw new ArgumentNullException(nameof(controlContext));
+
+            CheckBoxSelectAll.IsEnabled = ScreenshotAvailable;
+            AutomationProperties.SetName(lvResults, controlContext.DataGridAccessibleName);
```

Comment:
One way to do this in XAML would be to create a `DataGridAccessibleName` `DependencyProperty` in this class. You could then set `DataGridAccessibleName` in AutomatedChecksControl.xaml when you declare this control. A simple example of this pattern would be [`PlaceHolderTextBox`](https://github.com/microsoft/accessibility-insights-windows/blob/main/src/AccessibilityInsights.CommonUxComponents/Controls/PlaceholderTextBox.xaml.cs)--the property `Placeholder` is declared in the codebehind, [bound to in XAML (in the control's style in this case)](https://github.com/microsoft/accessibility-insights-windows/blob/f7b4c82e34341137333397731ffa9f80a4d29e74/src/AccessibilityInsights.SharedUx/Resources/Styles.xaml#L1557), and then set when [the control is used](https://github.com/microsoft/accessibility-insights-windows/blob/f7b4c82e34341137333397731ffa9f80a4d29e74/src/AccessibilityInsights.SharedUx/Controls/PropertyInfoControl.xaml#L37)